relay-cardinality/src/window.rs

```diff
@@ -33,20 +34,27 @@ impl SlidingWindow {
     /// * `request_timestamp / self.granularity_seconds - 2`
     /// * `request_timestamp / self.granularity_seconds - 3`
     /// * ...
-    pub fn iter(&self, timestamp: u64) -> impl Iterator<Item = u64> {
-        let value = timestamp / self.granularity_seconds;
+    pub fn iter(&self, timestamp: UnixTimestamp) -> impl Iterator<Item = Slot> {
+        let value = timestamp.as_secs() / self.granularity_seconds;
         (0..self.window_seconds / self.granularity_seconds)
             .map(move |i| value.saturating_sub(i + 1))
+            .map(Slot)
     }
 
     /// The active bucket is the oldest active granule.
-    pub fn active_slot(&self, timestamp: u64) -> u64 {
-        self.iter(timestamp).last().unwrap_or(0)
+    pub fn active_slot(&self, timestamp: UnixTimestamp) -> Slot {
+        self.iter(timestamp).last().unwrap_or(Slot(0))
     }
 }
 
+/// A single slot from a [`SlidingWindow`].
+#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Slot(pub u64);
```

Comment:
nit: Could we even make the `u64` private and instead implement `Display` for the redis key?